apps/site/assets/ts/stop/__tests__/components/amenities/AmenityCardTest.tsx

```diff
@@ -1,27 +1,47 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
-import AmenityCard from "../../../components/amenities/AmenityCard";
+import AmenityCard, {
+  AmenityModal
+} from "../../../components/amenities/AmenityCard";
 
 describe("AmenityCard", () => {
-  it("should render the icon & title", () => {
+  it("should render the icon, title, content", () => {
     render(
       <AmenityCard
         headerText={"Amenity Title"}
         icon={<span data-testid="icon"></span>}
-      />
+      >
+        <div>Card content</div>
+      </AmenityCard>
     );
     expect(screen.getByText("Amenity Title")).toBeDefined();
     expect(screen.getByTestId("icon")).toBeDefined();
+    expect(screen.getByText("Card content")).toBeDefined();
   });
 
-  it("should render children elements", () => {
+  it("should toggle children elements in a modal", () => {
     render(
-      <AmenityCard headerText={""} icon={<span></span>}>
-        <div>
-          <div>More content</div>
-        </div>
+      <AmenityCard
+        headerText={""}
+        icon={<span></span>}
+        modalContent={
+          <AmenityModal headerText="Inner header">
+            <div>
+              <div>Modal content</div>
+            </div>
+          </AmenityModal>
+        }
+      >
+        <div>Card content</div>
       </AmenityCard>
     );
-    expect(screen.getByText("More content")).toBeDefined();
+    expect(screen.queryByText("Inner header")).toBeNull();
+    expect(screen.queryByText("Modal content")).toBeNull();
+    screen.getByRole("button").click();
```

Comment:
suggestion: Thinking if another button was in the card this might not fetch the modal closing button. could this be fetched by className or an id added to the button?  Or maybe get the element by the close text